subsys/bluetooth/mesh/beacon.c

```diff
@@ -369,6 +372,7 @@ static void secure_beacon_recv(struct net_buf_simple *buf)
 	if (bt_mesh_beacon_enabled() &&
 	    sub->beacons_cur < 0xff) {
 		sub->beacons_cur++;
+		sub->beacon_recv = k_uptime_get_32();
```

Comment:
The use of 'k_uptime_get_32()' will create an issue, once the uptime for the node exceeds 1193 hours. At that time output of this function will roll over and will throw off this calculation because of overflow. I suggest for `beacon_recv` and `beacon_sent` timestamps 64 bit storage should be used along with regular k_uptime_get() call.

Reply:
With this being stored in a 32 bit variable, the overflow will sort itself out in the `time_diff_observed` calculation above, no?

Reply:
Umm, yes. It is always using 'now' to subtract smaller numbers, so this will work. If the above logic were to change or be reversed it will give the wrong outcome.